# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleMapKey.cpp
# Commit: 85fd9fc9cffa
# Full Hash: 85fd9fc9cffa98039328796d30a2d10386283b3e
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 2
# Description:
#   Bug 1742437 - Move module loader files into a central folder; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137532
# ==============================================================================

diff -r 7ecc4f9b20ad -r 85fd9fc9cffa dom/script/ModuleMapKey.cpp
--- a/dom/script/ModuleMapKey.cpp	Mon Feb 21 15:39:08 2022 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,38 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "ModuleMapKey.h"
-
-using namespace mozilla::dom;
-
-ModuleMapKey::ModuleMapKey(const nsIURI* aURI, nsIGlobalObject* aWebExtGlobal)
-    : nsURIHashKey(aURI), mWebExtGlobal(aWebExtGlobal) {}
-
-ModuleMapKey::ModuleMapKey(const ModuleMapKey* aKey)
-    : nsURIHashKey(aKey->mKey) {
-  *this = *aKey;
-}
-
-ModuleMapKey::ModuleMapKey(ModuleMapKey&& aToMove) noexcept
-    : nsURIHashKey(std::move(aToMove)) {
-  mWebExtGlobal = std::move(aToMove.mWebExtGlobal);
-}
-
-bool ModuleMapKey::KeyEquals(KeyTypePointer aOther) const {
-  if (mWebExtGlobal != aOther->mWebExtGlobal) {
-    return false;
-  }
-
-  return nsURIHashKey::KeyEquals(
-      static_cast<const nsURIHashKey*>(aOther)->GetKey());
-}
-
-// static
-PLDHashNumber ModuleMapKey::HashKey(KeyTypePointer aKey) {
-  PLDHashNumber hash = nsURIHashKey::HashKey(aKey->mKey);
-  // XXX This seems wrong.
-  hash =
-      AddToHash(hash, reinterpret_cast<uintptr_t>(aKey->mWebExtGlobal.get()));
-  return hash;
-}